pkg/intcode/intcode.go

```diff
@@ -29,51 +27,33 @@ const (
 )
 
 type program struct {
-	maxOp             opcode
-	index, input      int
-	runners           []runner
-	instructions, res []int
+	maxOp           opcode
+	index           int
+	inputs, outputs chan int
+	runners         []runner
+	instructions    []int
 }
 
 type runner interface {
 	run(p *program, arg1, arg2, arg3 int) bool
 }
 
-type add struct {
-}
+type add struct{}
+type multiply struct{}
+type input struct{}
+type output struct{}
+type jumpTrue struct{}
+type jumpFalse struct{}
+type less struct{}
+type equals struct{}
+type stop struct{}
 
-type multiply struct {
-}
-
-type input struct {
-}
-
-type output struct {
-}
-
-type jumpTrue struct {
-}
-
-type jumpFalse struct {
-}
-
-type less struct {
-}
-
-type equals struct {
-}
-
-type stop struct {
-}
-
-// Computer to compute a set of instructions.
-type Computer struct {
-}
-
-func newProgram(maxOp opcode, potentialInput *int, instructions []int) *program {
-	p := &program{
+func newProgram(maxOp opcode, inputs, outputs chan int, instructions []int) *program {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [line is 86 characters, out of limit 80](https://app.codacy.com/manual/afarbos/aoc/pullRequest?prid=5498712)